ci/checks/ensure-release-notes-up-to-date.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail -u
+
+current_version=$(grep '__version__ = ' git_machete/__init__.py | cut -d\' -f2)
+release_notes_version=$(egrep -o [0-9].[0-9].[0-9] RELEASE_NOTES.md -m 1)
+
+if [[ "$(printf "$current_version\n$release_notes_version" | sort --version-sort | tail -n 1)" != "$release_notes_version" ]]; then
```

Comment:
Hmmm isn't it enough to just compare `$current_version` and `$release_notes_version`? 🤔

Reply:
In `ensure-version-bumped.sh`, such logic makes sense, since there you're checking whether what's in `__init__.py` is **higher** than the latest git tag... but here only equality matters

Reply:
Yes, i though about that, this comparison also checks the case where `$release_notes_version` > `$current_version` but now since you mentioned it i am not so sure that this situation is correct, maybe it shouldn't be allowed, wdyt?

Reply:
Exactly, `$release_notes_version` being higher than `$current_version` should **NOT** be allowed.
Actually, some tooling (Debian changelog generation) even depends on the fact that the two are equal.